### DISCUSSION TITLE
Partial Data when batching entity fetches with errors

### DISCUSSION THREAD

**Discussion by kmoore-intuit** [Netflix/dgs-framework#1123] [General]
I noticed that when doing a batched entity fetch that is is an all or nothing call. So if I try to do an entity fetch where I attempt to retrieve an entity 3 times. Even if the first 2 calls succeeded, if the last entity fetch throws an error, the data variable in the response is empty/null with an error in the error portion. Is that intended? If so, why not give partial data? Also, why does the error not have the path for which one failed specifically?

**Comment by srinivasankavitha** [comment]
It should return partial data by design, so what you describe is not expected. Can you share any code snippets or a repro scenario so we can further investigate? Thanks!

**Reply by kmoore-intuit** [reply]
Hi,

Thanks for responding! That is definitely nice to know. I am working on a repo called graphql-orchestrator-java [graphql-orchestrator-java](https://github.com/intuit/graphql-orchestrator-java). It serves as a gateway for graphql services. From our gateway, we are sending this representation query to a downstream provider that is using dgs. This is the body for the request:
body: {
  "variables": {
      "REPRESENTATIONS": [
      {"__typename":"Movie" "movieId":"action movie"}
      {"__typename":"Movie" "movieId":"comedy movie"}
      {"__typename":"Movie" "movieId":"horror movie"}
      ]
  }
  "query":"query ($REPRESENTATIONS:[_Any!]!) {_entities(representations:$REPRESENTATIONS) {... on Movie {reviews {comment __typename}}}}"
}

In the following scenarios, here is the response from the provider using dgs version 4.1.0

1)  No error while during any of the entity fetches -> everything looks good 200 with data and nothing in the error
2) The first 2 entity are able to resolve and the third entity fetch throws exception ->  data is empty and error has the exception that was thrown and the path is empty too

**Reply by srinivasankavitha** [reply]
`4.1.0` is a very old version :) Could you try the latest, we might have fixed this issue along the way.

**Reply by kmoore-intuit** [reply]
Thanks updating to the latest version did solve the partial data problem. I did notice that the path in the error object is only _entities and does not give the actual path to the one that failed so for the case above I would expect it to show the path is 
_entities -> movie -> reviews[2] . Is that something you are intending to support in the near future?

**Reply by srinivasankavitha** [reply]
We don't have that information readily available at the level we are processing this, but something we can definitely explore as an improvement.
